src/storage/heap_file.c

```diff
@@ -24700,4 +24700,39 @@ heap_scancache::get_area_block_allocator ()
   return m_area->get_block_allocator ();
 }
 
+int
+heap_alloc_new_page (THREAD_ENTRY * thread_p, HFID * hfid, PGBUF_WATCHER * home_hint_p, VPID * new_page_vpid)
+{
+  int error_code = NO_ERROR;
+  HEAP_CHAIN new_page_chain;
+  VPID vpid;
+  PAGE_PTR page_ptr;
+
+  PGBUF_INIT_WATCHER (home_hint_p, PGBUF_ORDERED_HEAP_NORMAL, hfid);
+  
+  // Alloc a new page.
+  error_code = file_alloc (thread_p, &hfid->vfid, heap_vpid_init_new, &new_page_chain, &vpid, &page_ptr);
+  if (error_code != NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      return error_code;
+    }
+
+  // Need to get the watcher to the new page.
+  pgbuf_attach_watcher (thread_p, page_ptr, PGBUF_LATCH_WRITE, hfid, home_hint_p);
+
+  // Make sure we have fixed the page.
+  assert (pgbuf_is_page_fixed_by_thread (thread_p, &vpid));
+
+  VPID_COPY (&vpid, new_page_vpid);
+
+  return error_code;
+}
+
+int
+heap_get_default_empty_page_size ()
```

Comment:
Please rename for accuracy - `heap_nonheader_page_capacity ()` may be an option.